iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -51,6 +51,11 @@ private class TransientErrorStrategy : ITransientErrorDetectionStrategy
         {
             public bool IsTransient(Exception ex)
             {
+                if (ex is OperationCanceledException)
+                {
+                    return true;
+                }
```

Comment:
@drwill-ms I believe this is what we were looking for this morning. Now the retry policy will retry if the encountered error is an OperationCanceledException.

Reply:
I'm not sure we want this to be transient. If the user signals to cancel, but we say, ok sure but let me just retry first, that would be a problem. No?

Reply:
True. This may be a better fit for application level retry logic rather than SDK level retry logic.

Reply:
If you think this only happens when a ct is signalled, then yes. But I'm pretty sure the cases I ran into there was no cancellation token.

Reply:
I'll dig into MQTTNet more to see if there is some default timeout that they observe alongside the cancellationtoken we pass in, then. If they have an independent timeout, then I could see the SDK's retry logic marking those cases as retryable, sure.

Reply:
Looks like there is a default operation timeout of 100 seconds in MQTTNet regardless of cancellation token:

https://github.com/dotnet/MQTTnet/blob/ef85ef676a20fcd85ce1d6632053070d84c65381/Source/MQTTnet/Client/Options/MqttClientOptions.cs#L93

We can override this to TimeSpan.MaxValue though so that we only ever get MqttCommunicationTimedOutException when the cancellation token requests cancellation. With that, we can leave OperationCanceledException marked as a non-transient exception. Thoughts?